src/pj_transform.c

```diff
@@ -145,6 +136,7 @@ int pj_transform( PJ *srcdefn, PJ *dstdefn, long point_count, int point_offset,
                 {
                     x[point_offset*i] *= srcdefn->to_meter;
                     y[point_offset*i] *= srcdefn->to_meter;
+                    z[point_offset*i] *= srcdefn->to_meter;
```

Comment:
This looks dodgy. The vertical component should be scaled with `srcdefn->vto_meter`.

Consider this: 

```
cs2cs +proj=latlong +to +proj=utm +zone=32 +to_meter=0.5 +vto_meter=1.
```

With the current code both horizontal and vertical components will be scaled by 0.5.

Reply:
No, this corrects a legitimate bug - see issue #726 . Note that the scaling affected here is applied to geocentric (i.e. 3D ECEF cartesian) only, not to the utm coordinates in your example.

Reply:
Alright, it wasn't very clear in the convoluted diff. I see your point - this is the correct way to go.